src/plugins/index_pattern_editor/public/components/form_fields/timestamp_field.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { EuiFormRow, EuiComboBox, EuiFormHelpText, EuiComboBoxOptionOption } from '@elastic/eui';
+
+import {
+  UseField,
+  FieldConfig,
+  ValidationConfig,
+  getFieldValidityAndErrorMessage,
+} from '../../shared_imports';
+
+import { TimestampOption } from '../../types';
+import { schema } from '../form_schema';
+
+interface Props {
+  options: TimestampOption[];
+  isLoadingOptions: boolean;
+  isExistingIndexPattern: boolean;
+  isLoadingMatchedIndices: boolean;
+  hasMatchedIndices: boolean;
+}
+
+const requireTimestampOptionValidator = (options: Props['options']): ValidationConfig => ({
+  validator: async ({ value }) => {
+    const isValueRequired = !!options.length;
+    if (isValueRequired && !value) {
+      return {
+        message: i18n.translate(
+          'indexPatternEditor.requireTimestampOption.ValidationErrorMessage',
+          {
+            defaultMessage: 'Select a timestamp field option.',
```

Comment:
Can "option" be removed?

```suggestion
            defaultMessage: 'Select a timestamp field.',
```